reark/src/main/java/io/reark/reark/data/stores/cores/MemoryStoreCore.java

```diff
@@ -131,7 +130,11 @@ public MemoryStoreCore(@NonNull final Func2<U, U, U> putMergeFunction) {
 
             if (!valuesEqual) {
                 Log.v(TAG, "Merging values at " + id);
-                newItem = putMergeFunction.call(currentItem, newItem);
+                try {
+                    newItem = putMergeFunction.apply(currentItem, newItem);
+                } catch (Exception e) {
+                    Log.e(TAG, "Error merging!", e);
```

Comment:
Shouldn't this go to the `Single` as an error?

Reply:
Yes it should, done